ngraph/frontend/onnx_import/src/op/lstm.cpp

```diff
@@ -74,99 +125,171 @@ namespace ngraph
+                        // OpenVino Shape: [batch_size, seq_length, input_size]
+                        m_input_map[LSTMInput::LSTM_INPUT_X] =
                             builder::opset1::reorder_axes(ng_inputs.at(0), {1, 0, 2});
+
                         // Weight tensor for the gates.
                         // Shape: [num_directions, 4*hidden_size, input_size]
-                        m_map[LSTMInput::LSTM_INPUT_W] = ngraph::op::util::convert_lstm_node_format(
-                            ng_inputs.at(1),
-                            ngraph::op::util::LSTMWeightsFormat::IOFC,
-                            ngraph::op::util::LSTMWeightsFormat::FICO,
-                            1);
+                        m_input_map[LSTMInput::LSTM_INPUT_W] =
+                            ngraph::op::util::convert_lstm_node_format(
+                                ng_inputs.at(1),
+                                ngraph::op::util::LSTMWeightsFormat::IOFC,
+                                ngraph::op::util::LSTMWeightsFormat::FICO,
+                                1);
+
                         // The recurrence weight tensor.
                         // Shape: [num_directions, 4*hidden_size, hidden_size]
-                        m_map[LSTMInput::LSTM_INPUT_R] = ngraph::op::util::convert_lstm_node_format(
-                            ng_inputs.at(2),
-                            ngraph::op::util::LSTMWeightsFormat::IOFC,
-                            ngraph::op::util::LSTMWeightsFormat::FICO,
-                            1);
+                        m_input_map[LSTMInput::LSTM_INPUT_R] =
+                            ngraph::op::util::convert_lstm_node_format(
+                                ng_inputs.at(2),
+                                ngraph::op::util::LSTMWeightsFormat::IOFC,
+                                ngraph::op::util::LSTMWeightsFormat::FICO,
+                                1);
 
-                        const std::size_t hidden_size =
-                            m_map[LSTMInput::LSTM_INPUT_R].get_shape().back();
-                        const std::size_t batch_size =
-                            m_map[LSTMInput::LSTM_INPUT_X].get_shape().at(0);
-                        const std::size_t num_directions =
-                            m_map[LSTMInput::LSTM_INPUT_W].get_shape().front();
+                        // Check if creation of default inputs is needed
+                        auto ov_lstm_seqence_required_inputs_count = 7;
+                        if (ng_inputs.size() < ov_lstm_seqence_required_inputs_count)
+                        {
+                            init_dim_map();
+                        }
 
                         // ------ Optional inputs ------
-                        // The bias tensor for input gate. Shape [num_directions, 4*hidden_size]
+                        // `B` - The bias tensor for input gate. Shape [num_directions,
+                        // 8*hidden_size]
```

Comment:
4 * hidden_size is correct. B is the sum of biases

Reply:
`8 * hidden_size` is from [ONNX LSTM doc](https://github.com/onnx/onnx/blob/master/docs/Operators.md#LSTM
) 
`4 * hidden_size` is [OpsenVino LSTMSequence](https://github.com/openvinotoolkit/openvino/blob/master/docs/ops/sequence/LSTMSequence_1.md)

I've updated the comment to have in mind both interpretations

                        // `B` - The bias tensor for input gate.
                        // ONNX Shape: [num_directions, 8*hidden_size]
                        // OpenVino Shape: [num_directions, 4*hidden_size]